src/test/java/org/reaktivity/nukleus/maven/plugin/internal/generated/ListWithDefaultNullFW.java

```diff
@@ -0,0 +1,430 @@
+
+public class ListWithDefaultNullFW extends Flyweight
+{
+    private static final int PHYSICAL_LENGTH_OFFSET = 0;
+
+    private static final int PHYSICAL_LENGTH_SIZE = BitUtil.SIZE_OF_INT;
+
+    private static final int LOGICAL_LENGTH_OFFSET = PHYSICAL_LENGTH_OFFSET + PHYSICAL_LENGTH_SIZE;
+
+    private static final int LOGICAL_LENGTH_SIZE = BitUtil.SIZE_OF_INT;
+
+    private static final byte NULL_VALUE = 0x40;
+
+    private static final int FIRST_FIELD_OFFSET = LOGICAL_LENGTH_OFFSET + LOGICAL_LENGTH_SIZE;
+
+    private static final int INDEX_VARIANT_OF_STRING1 = 0;
+
+    private static final long MASK_VARIANT_OF_STRING1 = 1 << INDEX_VARIANT_OF_STRING1;
+
+    private static final int INDEX_VARIANT_OF_STRING2 = 1;
+
+    private static final long MASK_VARIANT_OF_STRING2 = 1 << INDEX_VARIANT_OF_STRING2;
+
+    private static final int INDEX_VARIANT_OF_UINT = 2;
+
+    private static final long MASK_VARIANT_OF_UINT = 1 << INDEX_VARIANT_OF_UINT;
+
+    private static final int INDEX_VARIANT_OF_INT = 3;
+
+    private static final long MASK_VARIANT_OF_INT = 1 << INDEX_VARIANT_OF_INT;
+
+    private static final long DEFAULT_VARIANT_OF_UINT = 4000000000L;
+
+    private static final int NULL_VALUE_SIZE = BitUtil.SIZE_OF_BYTE;
+
+    private VariantEnumKindWithString32FW variantOfString1RO = new VariantEnumKindWithString32FW();
+
+    private VariantEnumKindWithString32FW variantOfString2RO = new VariantEnumKindWithString32FW();
+
+    private VariantEnumKindOfUint32FW variantOfUintRO = new VariantEnumKindOfUint32FW();
+
+    private VariantEnumKindWithInt32FW variantOfIntRO = new VariantEnumKindWithInt32FW();
+
+    private long bitmask;
+
+    public int length()
+    {
+        return buffer().getInt(offset() + LOGICAL_LENGTH_OFFSET);
+    }
+
+    public String variantOfString1()
+    {
+        assert (bitmask & MASK_VARIANT_OF_STRING1) != 0L : "Field \"variantOfString1\" is not set";
+        return variantOfString1RO.get();
+    }
+
+    public String variantOfString2()
+    {
+        assert (bitmask & MASK_VARIANT_OF_STRING2) != 0L : "Field \"variantOfString2\" is not set";
+        return variantOfString2RO.get();
+    }
+
+    public long variantOfUint()
+    {
+        return (bitmask & MASK_VARIANT_OF_UINT) != 0L ? variantOfUintRO.get() : DEFAULT_VARIANT_OF_UINT;
+    }
+
+    public int variantOfInt()
+    {
+        assert (bitmask & MASK_VARIANT_OF_INT) != 0L : "Field \"variantOfInt\" is not set";
+        return variantOfIntRO.get();
+    }
+
+    @Override
+    public ListWithDefaultNullFW wrap(
+        DirectBuffer buffer,
+        int offset,
+        int maxLimit)
+    {
+        super.wrap(buffer, offset, maxLimit);
+        checkLimit(offset + PHYSICAL_LENGTH_OFFSET + PHYSICAL_LENGTH_SIZE, maxLimit);
+        final int limit = limit();
+        checkLimit(limit, maxLimit);
+        checkLimit(offset + LOGICAL_LENGTH_OFFSET + LOGICAL_LENGTH_SIZE, limit);
```

Comment:
This `checkLimit` for logical length is probably not necessary if you already have the full` limit` check above, right?